framework_tvb/tvb/tests/framework/conftest.py

```diff
@@ -268,68 +275,240 @@ def build(connectivity=None, simulation_length=100):
-        covariance_db.fk_from_operation = op.id
-        covariance_db.fill_from_has_traits(covariance)
+        data_type_index = DummyDataTypeIndex(subject=subject)
+        data_type_index.fk_from_operation = operation.id
+        data_type_index.fill_from_has_traits(data_type)
 
-        covariance_h5_path = h5.path_for_stored_index(covariance_db)
-        with TimeSeriesH5(covariance_h5_path) as f:
-            f.store(ts)
+        data_type_h5_path = h5.path_for_stored_index(data_type_index)
+        with DummyDataTypeH5(data_type_h5_path) as f:
+            f.store(data_type)
 
-        session.add(covariance_db)
-        session.commit()
-        return covariance_db
+        data_type_index = dao.store_entity(data_type_index)
+        return data_type_index
 
     return build
+
+
+dummy_datatype_index_factory2 = dummy_datatype_index_factory
+
+dummy_datatype_index_factory3 = dummy_datatype_index_factory
+
+
+@pytest.fixture()
+def datatype_measure_factory(operation_factory):
+    def build(analyzed_entity):
+
+        measure = DatatypeMeasureIndex()
+        measure.metrics = '{"v": 3}'
+        measure.source = analyzed_entity
+        measure = dao.store_entity(measure)
+
+        return measure
+
+    return build
+
+
+@pytest.fixture()
+def datatype_group_factory(time_series_index_factory, datatype_measure_factory, project_factory, user_factory, operation_factory):
+    def build(subject="Datatype Factory User", state="RAW_DATA", project=None):
+
+        range_1 = ["row1", [1, 2, 3]]
+        range_2 = ["row2", [0.1, 0.3, 0.5]]
+
+        user = user_factory()
+
+        if project is None:
+            project = project_factory(user)
+
+        # Create an algorithm
+        alg_category = AlgorithmCategory('one', True)
+        dao.store_entity(alg_category)
+        ad = Algorithm(IntrospectionRegistry.SIMULATOR_MODULE, IntrospectionRegistry.SIMULATOR_CLASS,
+                       alg_category.id)
+        algorithm = dao.get_algorithm_by_module(IntrospectionRegistry.SIMULATOR_MODULE,
+                                                IntrospectionRegistry.SIMULATOR_CLASS)
+
+        if algorithm is None:
+            algorithm = dao.store_entity(ad)
+
+        # Create meta
+        meta = {DataTypeMetaData.KEY_SUBJECT: "Datatype Factory User",
+                DataTypeMetaData.KEY_STATE: "RAW_DATA"}
+
+        # Create operation
+        operation = operation_factory(algorithm=algorithm, test_user=user, test_project=project, meta=meta)
+
+        group = OperationGroup(project.id, ranges=[json.dumps(range_1), json.dumps(range_2)])
+        group = dao.store_entity(group)
+        group_ms = OperationGroup(project.id, ranges=[json.dumps(range_1), json.dumps(range_2)])
+        group_ms = dao.store_entity(group_ms)
+
+        datatype_group = DataTypeGroup(group, subject=subject, state=state, operation_id=operation.id)
+        # datatype_group.storage_path = self.files_helper.get_project_folder(project, str(operation.id))
```

Comment:
storage_path is no longer available on datatypes. Search for all places doing this and fix them.